File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/parts/resource/EnergyBarTest.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.IntSteps;
import com.bonepl.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class EnergyBarTest extends AbstractResourceTest {
    @Test
    void testEnergyBar() {
        final IntSteps intSteps = new IntSteps(0, 100, 5);
        final EnergyBar energyBar = new EnergyBar();

        new AnimationTester()
                .withBeforeIterationAction(i -> mockResource(intSteps.nextInt(), 100))
                .testAnimation(energyBar, 40);
    }
}

File: test/hud/parts/resource/EnergyBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {EnergyBar} from '../../../../src/hud/parts/resource/EnergyBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for energy resource bar on actual hardware. */
describe('Energy Integration', () => {
    setupResourceTest();

    test('testEnergyBar', async () => {
        const intSteps = new IntSteps(0, 100, 5);
        const energyBar = new EnergyBar();

        await new AnimationTester()
            .withBeforeIterationAction(() => mockResource(intSteps.nextInt(), 100))
            .testAnimation(energyBar, 40);
    }, 30000);
});
